udf/mrn_udf_highlight_html.cpp

```diff
@@ -286,13 +304,40 @@ MRN_API mrn_bool mroonga_highlight_html_init(UDF_INIT *init,
   }
 
   info->query_mode.used = false;
+  info->specify_tag.used = false;
+
+  for (unsigned int i = 0; i < args->arg_count; i++) {
+    const char *attribute = args->attributes[i];
+    unsigned long attribute_length = args->attribute_lengths[i];
+    if (args->attribute_lengths[i] == strlen("query") &&
+        strncmp(args->attributes[i], "query", strlen("query")) == 0) {
+      info->query_mode.used = true;
+      info->query_mode.table = NULL;
+      info->query_mode.default_column = NULL;
+      info->query_mode.query = args->args[i];
+      info->query_mode.query_length = args->lengths[i];
+    }
+    if (attribute_length == strlen("open_tag") &&
+        strncmp(attribute, "open_tag", strlen("open_tag")) == 0) {
+      info->specify_tag.used = true;
+      info->specify_tag.open_tag = args->args[i];
+    } else if (attribute_length == strlen("close_tag") &&
+               strncmp(attribute, "close_tag", strlen("close_tag")) == 0) {
+      info->specify_tag.used = true;
+      info->specify_tag.close_tag = args->args[i];
+    }
+  }
 
-  if (args->arg_count == 2 &&
-      args->attribute_lengths[1] == strlen("query") &&
-      strncmp(args->attributes[1], "query", strlen("query")) == 0) {
-    info->query_mode.used = true;
-    info->query_mode.table = NULL;
-    info->query_mode.default_column = NULL;
+  if (info->specify_tag.used) {
+    if (info->specify_tag.open_tag == nullptr) {
+      snprintf(message, MYSQL_ERRMSG_SIZE,
+       "mroonga_highlight_html(): missing the required \"open_tag\" argument.");
+      goto error;
+    } else if (info->specify_tag.close_tag == nullptr) {
+      snprintf(message, MYSQL_ERRMSG_SIZE,
+       "mroonga_highlight_html(): missing the required \"close_tag\" argument.");
+      goto error;
+    } 
```

Comment:
How about using `<span class="keyword">` and `</span>` as the default?

Reply:
Fixed.